kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -126,6 +138,15 @@ public static String uid(@NonNull HasMetadata resource) {
         return resource.getMetadata().getUid();
     }
 
+    public static String coordinatesString(LocalRef<?> ref, HasMetadata primary) {
+        GroupKind groupKind = new GroupKind(ref.getGroup(), ref.getKind());
+        String plural = GROUP_KIND_TO_PLURAL.get(groupKind);
```

Comment:
maybe better to go for the singular and just lowercase the kind in the LocalRef to avoid this mapping

Reply:
I'm not aware of any counterexamples to that working.